### PR TITLE
docs (external-data): Update README with pattern strategies and surface level api

### PR DESCRIPTION
**Pull request by ghoshkaj**
Adding documentation to the external-data example app's readme.

### CODE REVIEW

File: examples/hosts/app-integration/external-data/README.md [hunk 1]
```diff
@@ -6,15 +6,54 @@ Please note that the ideas explored here are experimental and under development.
 
 ## Scenario
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:7]: Be sure to run `npm run format` from within the package. I suspect prettier will want to upate some formatting here :)

File: examples/hosts/app-integration/external-data/README.md [hunk 2]
```diff
@@ -6,15 +6,54 @@ Please note that the ideas explored here are experimental and under development.
...
 
-TBD
+This example demonstrates a scenario in which the Customers "source of truth" of their data lives in a service that is external to the Fluid service. The Customers can then:
+1. Import the external data into a Fluid collboration session.
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:10]: Nit
```suggestion
1. Import the external data into a Fluid collaboration session.
```

File: examples/hosts/app-integration/external-data/README.md [hunk 3]
```diff
@@ -6,15 +6,54 @@ Please note that the ideas explored here are experimental and under development.
...
+2. Export data from a Fluid collaboration session back to the source of truth.
+3. Sync updates between Fluid and the source of truth in as close to real-time as the scenario allows.
+
+In this case, the Fluid collaboration session serves as a "drafting surface" in which clients collaborate to create a draft of the data and then send the saved data back to the source of truth for long term storage.
 
 ## Strategy overview
 
-TBD
+Many services that would hold the "source of truth" data offer explicit commit style interfaces (e.g. vi REST call or similar) which are not well suited to rapid updates. However, they often expose third-party integration via REST APIS for querying and manipulating data, as well as webhooks for watching updates to the data.
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:18]: (Just a suggestion, feel free to disregard) github style Markdown is spec'd to treat adjacent lines (separated by exactly 1 newline) as sentences in the same paragraph. When rendered, they will be separated by a single whitespace character. I like to recommend breaking lines in Markdown docs along sentence boundaries, as it makes future git-wise diffs easier to reason about (you only have to consider changes on a sentence-by-sentence basis).

Comment (Josmithr) [reply examples/hosts/app-integration/external-data/README.md:18]:
```suggestion
Many services that would hold the "source of truth" data offer explicit commit style interfaces (e.g. vi REST call or similar) which are not well suited to rapid updates.
However, they often expose third-party integration via REST APIS for querying and manipulating data, as well as webhooks for watching updates to the data.
```

File: examples/hosts/app-integration/external-data/README.md [hunk 4]
```diff
@@ -6,15 +6,54 @@ Please note that the ideas explored here are experimental and under development.
...
+
+This repo contains an external service that serves as a mock external "source of truth" data server, that offers this REST API collection and webhook interfaces in ./src/mock-external-data-service. The APIs served by this external service are the following:
+
+1. POST `/register-for-webhook`
+2. GET `/fetch-tasks`
+3. POST `/set-tasks`
+4. POST `/debug-reset-task-list`
+
+Find the documentation for them in the code itself: (./src/mock-external-data-service/service.ts)[./src/mock-external-data-service/service.ts]
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:27]: Might be worth pointing to the service's README file instead (we can document the individual routes there and point to the code from there).

File: examples/hosts/app-integration/external-data/README.md [hunk 5]
```diff
@@ -6,15 +6,54 @@ Please note that the ideas explored here are experimental and under development.
...
+
+Next we need a service that will register the webhooks and listen for incoming changes. In a true implementation, this registration would happen in the Fluid server-side, potentially in the Alfred service, as we have in our dev branch here: https://github.com/microsoft/FluidFramework/blob/dev/external-data-prototyping/server/routerlicious/packages/lambdas/src/alfred/index.ts#L463-L508.
+
+However, for the purposes of this example, we have built out a separate service to register for the webhook. It contains the following endpoints:
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:31]: I think this service exists even with the Alfred changes. I think the idea was that Alfred (or whatever realistic Fluid Service scenario we end up using) registers with the app's custom service, and not with the external data service directly.

File: examples/hosts/app-integration/external-data/README.md [hunk 6]
```diff
@@ -6,15 +6,54 @@ Please note that the ideas explored here are experimental and under development.
...
+1. POST `/register-for-webhook`
+2. POST `/echo-external-data-webhook`
+
+Find the details in the code here: (examples/hosts/app-integration/external-data/src/mock-customer-service/service.ts)[examples/hosts/app-integration/external-data/src/mock-customer-service/service.ts]
+
+Next we come to how the Fluid collaboration session and clients will consume the APIs above and render them to the screen. To accomplish that, this example repo explores two routes to implement the scenario above. One is the Echo Webhook Pattern and the other is the Bot Pattern. Both are documented in more detail below.
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:37]: Nit
```suggestion
Next we come to how the Fluid collaboration session and clients will consume the APIs above and render them to the screen. To accomplish that, this example app explores two routes to implement the scenario above. One is the Echo Webhook Pattern and the other is the Bot Pattern. Both are documented in more detail below.
```

File: examples/hosts/app-integration/external-data/README.md [hunk 7]
```diff
@@ -6,15 +6,54 @@ Please note that the ideas explored here are experimental and under development.
...
 ### Echo Webhook Pattern
 
-TODO: Document the echo-webhook pattern and how it is surfaced in the example code.
+<img width="1356" alt="image" src="https://user-images.githubusercontent.com/6777404/216415477-14d0b193-29c9-48e5-8b6b-0a549a5dde58.png">
+
+<img width="1404" alt="image" src="https://user-images.githubusercontent.com/6777404/216417448-2a43db3e-12a2-48a6-b6d0-a6c4a95e27d1.png">
+
+<img width="1374" alt="image" src="https://user-images.githubusercontent.com/6777404/216417779-12861504-7909-489c-b7a2-4d75814a396f.png">
+
+In this pattern. the indivdual clients are responsible for authenticatin with the external data source and making REST calls as needed. However, since the the "customer server" from the section prior, is registered to the webhooks and listening for incoming changes, the clients do not have a way to know that there has been a change upstream. The "echo webhook" pattern is therefore a signal that the server side broadcasts to the Fluid service to echo the information that it has received from the webhook that there is incoming information.
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:47]:
```suggestion
In this pattern. the individual clients are responsible for authentication with the external data source and making REST calls as needed. However, since the "customer server" from the section prior is registered to the webhooks and listening for incoming changes, the clients do not have a way to know that there has been a change upstream. The "echo webhook" pattern is therefore a signal that the server side broadcasts to the Fluid service to echo the information that it has received from the webhook that there is incoming information.
```

File: examples/hosts/app-integration/external-data/README.md [hunk 8]
```diff
@@ -6,15 +6,54 @@ Please note that the ideas explored here are experimental and under development.
...
+
+The clients (or elected leader client) can then send a fetch call to retrieve the information and dsiplay it to screen by making a call to the external data server's GET `/fetch-tasks` endpoint.
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:49]:
```suggestion
The clients (or elected leader client) can then send a fetch call to retrieve the information and display it to screen by making a call to the external data server's GET `/fetch-tasks` endpoint.
```

**Review by Josmithr — COMMENTED** [review]
Thank you for writing this up! This is great! I left a couple of comments / suggestions, but I think the content looks great!

File: examples/hosts/app-integration/external-data/README.md [hunk 9]
```diff
@@ -6,15 +6,55 @@ Please note that the ideas explored here are experimental and under development.
 
 ## Scenario
 
-TBD
+This example demonstrates a scenario in which the Customers "source of truth" of their data lives in a service that is external to the Fluid service. The Customers can then:
```
Comment (WayneFerrao) [inline examples/hosts/app-integration/external-data/README.md:9]: Nit: I feel like this reads a bit better.
```suggestion
This example demonstrates a scenario in which the "source of truth" of Customer data lives in a service that is external to the Fluid service. The Customers can then:
```

File: examples/hosts/app-integration/external-data/README.md [hunk 10]
```diff
@@ -6,15 +6,55 @@ Please note that the ideas explored here are experimental and under development.
...
+3. Sync updates between Fluid and the source of truth in as close to real-time as the scenario allows.
+
+In this case, the Fluid collaboration session serves as a "drafting surface" in which clients collaborate to create a draft of the data and then send the saved data back to the source of truth for long term storage.
 
 ## Strategy overview
 
-TBD
+Many services that would hold the "source of truth" data offer explicit commit style interfaces (e.g. vi REST call or similar) which are not well suited to rapid updates. However, they often expose third-party integration via REST APIS for querying and manipulating data, as well as webhooks for watching updates to the data.
+
+This repo contains an external service that serves as a mock external "source of truth" data server, that offers this REST API collection and webhook interfaces in ./src/mock-external-data-service. The APIs served by this external service are the following:
```
Comment (WayneFerrao) [inline examples/hosts/app-integration/external-data/README.md:21]: Just a suggested edit, which I feel reads easier to someone who doesn't have any context.
```suggestion
This repo contains a service that mocks the external "source of truth" data server. This mock service offers a REST API collection and webhook interfaces in ./src/mock-external-data-service. The API requests served by this "external" service are the following:
```

File: examples/hosts/app-integration/external-data/README.md [hunk 11]
```diff
@@ -6,15 +6,55 @@ Please note that the ideas explored here are experimental and under development.
...
+
+Upon receipt of new external data, the external data is written immediately into the "SavedData" map, and a check occurs comparing the SavedData to the DraftData. If there are changes between the two, these are displayed on screen and the clients can (currently) only choose to consume the changes and overwrite their local data, via using regular ops mechanism. The first person to overwrite will use regular ops to write into the FluidData and the change will be attributed to them.
+
+The collaboration session can continue as expected, and when the collboartion session is ready to be closed, the clients can simply Save Changes to write back to the External Data Source by making a request to the External Data Server's POST `/set-tasks` endpoint.
```
Comment (WayneFerrao) [inline examples/hosts/app-integration/external-data/README.md:57]: Nit 
```suggestion
The collaboration session can continue as expected, and when the collaboration session is ready to be closed, the clients can simply Save Changes to write back to the External Data Source by making a request to the External Data Server's POST `/set-tasks` endpoint.
```

File: examples/hosts/app-integration/external-data/README.md (hunk 5 above)
Comment (ChumpChief) [reply examples/hosts/app-integration/external-data/README.md:33]: At minimum there's a translation layer - the external data is in some format that needs to be translated into a thing that Fluid can understand (e.g. turn some HTTP response into a Signal).  Also auth considerations - the Fluid service shouldn't pick up a responsibility of authenticating to arbitrary external data sources on our customers' behalf.

Later, for the bot pattern this customer service will also be where the bot runs most likely.

Comment (ghoshkaj) [reply examples/hosts/app-integration/external-data/README.md:33]: Aha okay, I get it now. Will re-write given this new info!

File: examples/hosts/app-integration/external-data/README.md [hunk 12]
```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
...
 
-TBD
+In order to accomplish the goals above, we have split up the responsibilities into a few different pieces:
+
+**External Data Service**
+
+Many services that would hold the "source of truth" data offer explicit commit style interfaces (e.g. vi REST call or similar) which are not well suited to rapid updates.
+However, they often expose third-party integration via REST APIS for querying and manipulating data, as well as webhooks for watching updates to the data.
+
+This repo contains a service that mocks the external "source of truth" data server. This mock service offers a REST API collection and webhook interfaces in ./src/mock-external-data-service. The API requests served by this "external" service are the following:
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:26]: Nit ```suggestion
This repo contains a service that mocks the external "source of truth" data server. This mock service offers a REST API collection and webhook interfaces in `./src/mock-external-data-service`. The API requests served by this "external" service are the following:
```

File: examples/hosts/app-integration/external-data/README.md [hunk 13]
```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
...
+2. POST `/echo-external-data-webhook`: "Echoes" the external data services data update notifications to our own webhook subscribers. This data will be forwarded to our own subscribers.
+
+Find the details of the API in the [Customer Service README](./src/mock-customer-service/README.md).
+
+Next we come to how the Fluid collaboration session and clients will consume the APIs above and render the data to the screen. This example repo explores two routes to accomplish this. One is the Echo Webhook Pattern and the other is the Bot Pattern. Both are documented in more detail below.
 
 ### Echo Webhook Pattern
 
-TODO: Document the echo-webhook pattern and how it is surfaced in the example code.
+<img width="1356" alt="image" src="https://user-images.githubusercontent.com/6777404/216415477-14d0b193-29c9-48e5-8b6b-0a549a5dde58.png">
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:50]: Just curious: what is the lifetime of these images? Should we check them in alongside this documentation instead?

Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:50]: Nit: For the `alt` property of these images, we should probably differentiate between the different diagrams rather than labeling them all as "image".

File: examples/hosts/app-integration/external-data/README.md [hunk 14]
```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
...
+
+<img width="1404" alt="image" src="https://user-images.githubusercontent.com/6777404/216417448-2a43db3e-12a2-48a6-b6d0-a6c4a95e27d1.png">
+
+<img width="1374" alt="image" src="https://user-images.githubusercontent.com/6777404/216417779-12861504-7909-489c-b7a2-4d75814a396f.png">
+
+Since the Customer Service is registered to the webhooks and listening for incoming changes, the clients do not have a way to know that there has been a change upstream.
+
+The "echo webhook" pattern accomplishes this by having the Fluid service subscribe to "echo webhooks" in the Customer Service, listening for the webhook notifications there, and then sending a Signal to the clients to alert them of this upstream change.
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:58]: I would maybe reword this to something like "... the Fluid service subscribe to notifications from the Customer Service when it receives webhook notifications from the external data provider...". "Echo Webhook" is the name we've given the pattern here, but I think being more descriptive is probably useful.

File: examples/hosts/app-integration/external-data/README.md [hunk 15]
```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
...
+1. POST `/register-for-webhook`: Register's the sender's URL (in this case the Customer Service's URL) to receive notifications when the external task-list data changes.
+2. GET `/fetch-tasks`: Fetches the task list from the external data store.
+3. POST `/set-tasks`: Updates external data store with new tasks list (complete override).
+4. POST `/debug-reset-task-list`: Resets the external data to its original contents.
+
+Find the details of the API in the [External Data Service README](./src/mock-external-data-service/README.md)
+
+**Customer Service**
+
+Next we need a customer service that functions as the intermediary between the External Data Service and the Fluid Service. This server is responsible for authenticating to the external service on the customer's behalf. It registers to the External Data Service webhooks and listens for incoming changes. It also acts as a translation layer, tasnlating to and from the External Data Service expected format and the Fluid Service's expected format.
```
Comment (ChumpChief) [inline examples/hosts/app-integration/external-data/README.md:37]:
```suggestion
Next we need a customer service that functions as the intermediary between the External Data Service and the Fluid Service. This server is responsible for authenticating to the external service on the customer's behalf. It registers to the External Data Service webhooks and listens for incoming changes. It also acts as a translation layer, translating to and from the External Data Service expected format and the Fluid Service's expected format.
```

File: examples/hosts/app-integration/external-data/README.md [hunk 16]
```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
...
+
+<img width="1374" alt="image" src="https://user-images.githubusercontent.com/6777404/216417779-12861504-7909-489c-b7a2-4d75814a396f.png">
+
+Since the Customer Service is registered to the webhooks and listening for incoming changes, the clients do not have a way to know that there has been a change upstream.
+
+The "echo webhook" pattern accomplishes this by having the Fluid service subscribe to "echo webhooks" in the Customer Service, listening for the webhook notifications there, and then sending a Signal to the clients to alert them of this upstream change.
+
+So, on receiving the notification from the External Server's webhook, the Customer Service makes a POST request using the `/echo-external-data-webhook` which simply echoes the information to the Fluid Service.
+
+On receiving notification from the Echo webhhooks, Alfred (the Fluid Service Front End) broadcasts a Signal to let the the clients know that there has been a changeupstream.
```
Comment (ChumpChief) [inline examples/hosts/app-integration/external-data/README.md:62]:
```suggestion
On receiving notification from the Echo webhooks, Alfred (the Fluid Service Front End) broadcasts a Signal to let the the clients know that there has been a change upstream.
```

File: examples/hosts/app-integration/external-data/README.md (hunk 14 above)
Comment (Josmithr) [reply examples/hosts/app-integration/external-data/README.md:58]: Same with subsequent references to "echo webhooks" - I think describing the scenarios in terms of subscription / notifications will probably be easier readers to understand.

File: examples/hosts/app-integration/external-data/README.md [hunk 17]
```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
...
+
+We have a prototype of the echo webhhook and signal broadcast in Alfred in our dev branch here: https://github.com/microsoft/FluidFramework/blob/dev/external-data-prototyping/server/routerlicious/packages/lambdas/src/alfred/index.ts#L463-L508.
```
Comment (ChumpChief) [inline examples/hosts/app-integration/external-data/README.md:64]: This statement will become stale as soon as a change is made - maybe consider phrasing and a link that could be a bit more durable?

File: examples/hosts/app-integration/external-data/README.md [hunk 18]
```diff
@@ -24,6 +77,22 @@ TODO: Document the bot pattern and how it is surfaced in the example code.
 
 TBD
 
+### Concepts of data in this repository
+
+A few useful concepts to understand in implementing conflict resolution in this app:
+
+Task - This is the unit that can be edited and attributed to an author. It is also the level at which conflict resolution currently takes place. The task holds knowledge of the local edits as well as the external edits and can display one or both to the screen.
+
+TaskList - This can be compared to a "board" that holds all of the tasks. It is the larger visible entity to show up in a component within the Loop app, or any similar app. The app stores two types of TaskList's - a "draft" version and a "saved version". More on these below.
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:86]: I probably wouldn't refer to Loop specifically here. I would just refer to "the application" in the abstract.

```suggestion
TaskList - This can be compared to a "board" that holds all of the tasks. It is the larger visible entity to show up in a component within the Loop app, or any similar app. The app stores two types of TaskList's - a "draft" version and a "saved version". More on these below.
```

File: examples/hosts/app-integration/external-data/README.md [hunk 19]
```diff
@@ -24,6 +77,22 @@ TODO: Document the bot pattern and how it is surfaced in the example code.
...
+
+SavedData - Data that comes in fresh from the external data source is first stored in a SharedMap known as "SavedData".
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:88]: Nit: Might be nice to link to our SharedMap docs, for readers who may be unfamiliar with it.
```suggestion
SavedData - Data that comes in fresh from the external data source is first stored in a [SharedMap](https://fluidframework.com/docs/data-structures/map/) known as "SavedData".
```

File: examples/hosts/app-integration/external-data/README.md [hunk 20]
```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
...
+In order to accomplish the goals above, we have split up the responsibilities into a few different pieces:
+
+**External Data Service**
+
+Many services that would hold the "source of truth" data offer explicit commit style interfaces (e.g. vi REST call or similar) which are not well suited to rapid updates.
+However, they often expose third-party integration via REST APIS for querying and manipulating data, as well as webhooks for watching updates to the data.
+
+This repo contains a service that mocks the external "source of truth" data server. This mock service offers a REST API collection and webhook interfaces in ./src/mock-external-data-service. The API requests served by this "external" service are the following:
+
+1. POST `/register-for-webhook`: Register's the sender's URL (in this case the Customer Service's URL) to receive notifications when the external task-list data changes.
```
Comment (ChumpChief) [inline examples/hosts/app-integration/external-data/README.md:28]: Maybe add a note for each of these who we expect to call them (customer's service vs. fluid client)

File: examples/hosts/app-integration/external-data/README.md [hunk 21]
```diff
@@ -24,6 +77,22 @@ TODO: Document the bot pattern and how it is surfaced in the example code.
 
 TBD
 
+### Concepts of data in this repository
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:80]: High-level feedback for this section: I don't know how much we need to refer to implementation details here. E.g. the fact that we use SharedMap_s to store the Draft / Saved data. It might be better to just describe what they represent, and let the reader explore the code if they want to see the particular implementation details.

@ChumpChief interested in your thoughts on this.

File: examples/hosts/app-integration/external-data/README.md [hunk 22]
```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
...
+
+Find the details of the API in the [External Data Service README](./src/mock-external-data-service/README.md)
+
+**Customer Service**
+
+Next we need a customer service that functions as the intermediary between the External Data Service and the Fluid Service. This server is responsible for authenticating to the external service on the customer's behalf. It registers to the External Data Service webhooks and listens for incoming changes. It also acts as a translation layer, tasnlating to and from the External Data Service expected format and the Fluid Service's expected format.
+
+In this example, the Customer Service contains the following endpoints:
+
+1. POST `/register-for-webhook`: Register's the sender's URL (in this case, the Fluid Service's URL) to receive notifications when the external task-list data changes.
```
Comment (ChumpChief) [inline examples/hosts/app-integration/external-data/README.md:41]: This doesn't sound right -- I don't think the customer's service provides any webhooks to be registered for, but instead it registers for webhooks from both the external service and the fluid service?

File: examples/hosts/app-integration/external-data/README.md [hunk 23]
```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
...
+2. POST `/echo-external-data-webhook`: "Echoes" the external data services data update notifications to our own webhook subscribers. This data will be forwarded to our own subscribers.
```
Comment (ChumpChief) [inline examples/hosts/app-integration/external-data/README.md:42]: I think this is the URL that gets registered for the external service's webhook, is that right?

File: examples/hosts/app-integration/external-data/README.md (hunk 21 above)
Comment (ChumpChief) [reply examples/hosts/app-integration/external-data/README.md:80]: Yeah - probably just go to the level of detail here so the reader can see all the puzzle pieces fitting together, which is prob no deeper than the public API surface.  For deeper documentation, probably put it in the class TSDoc for the respective classes.

**Review by ChumpChief — APPROVED** [review]
Looking good!  Approving preemptively -- any further improvements you make before checking in are great, or could be done as followups I think (I expect some of this will get tweaked over time as the example is worked on anyway).

File: examples/hosts/app-integration/external-data/README.md (hunk 13 above)
Comment (ghoshkaj) [reply examples/hosts/app-integration/external-data/README.md:50]: I think the lifetime of these images is up until the point that we and ODSP and FRS agree on one mechanism and we + they implement it so I'm thinking like a few months but not ad infinitum? If we check them in alongside this documentation, is that better for short term or long term lifetime?

Comment (ghoshkaj) [reply examples/hosts/app-integration/external-data/README.md:50]: Good point, I'll update them!

File: examples/hosts/app-integration/external-data/README.md (hunk 21 above)
Comment (ghoshkaj) [reply examples/hosts/app-integration/external-data/README.md:80]: Sounds good! I'll take the details out!

File: examples/hosts/app-integration/external-data/README.md (hunk 23 above)
Comment (Josmithr) [reply examples/hosts/app-integration/external-data/README.md:42]: It is, yes. And I think this should be considered an implementation detail of the customer service, so we probably shouldn't add documentation here about it.

File: examples/hosts/app-integration/external-data/README.md (hunk 22 above)
Comment (Josmithr) [reply examples/hosts/app-integration/external-data/README.md:41]: This describes the current pattern, but it probably isn't what we want. What you describe makes a lot more sense, so we should update the customer pattern to know the Fluid service endpoint, and not require any sort of registration from the Fluid service before beginning to echo incoming updates.

File: examples/hosts/app-integration/external-data/README.md [hunk 24]
```diff
@@ -6,15 +6,62 @@ Please note that the ideas explored here are experimental and under development.
...
+
+**Customer Service**
+
+Next we need a customer service that functions as the intermediary between the External Data Service and the Fluid Service. This server is responsible for authenticating to the external service on the customer's behalf. It registers to the External Data Service webhooks and listens for incoming changes. It also acts as a translation layer, translating to and from the External Data Service expected format and the Fluid Service's expected format.
+
+In this example, the Customer Service contains the following endpoints:
+
+1. POST `/register-for-webhook`: Registers the sender's URL to receive notifications when the external task-list data changes. Currently, Alfred registers it's `/task-list-hook` endpoint to this URL to be called when the Customer Service receives a notification on its `/echo-external-data-webhook` endpoint.
+2. POST `/echo-external-data-webhook`: "Echoes" the external data services data update notifications to our own webhook subscribers. This data will be forwarded to our own subscribers. Registered to be called by the External Data Service when there's been a change to data upstream.
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/README.md:42]: I still think we should remove this from the documentation here. It should be considered a private implementation detail of the customer service.

**Review by Josmithr — APPROVED** [review]

File: examples/hosts/app-integration/external-data/README.md (hunk 24 above)
Comment (ghoshkaj) [reply examples/hosts/app-integration/external-data/README.md:42]: Thinking about this more, I don't think we should remove it because this example app is to prototype the full flow. It's role is to provide a skeleton of how a customer would eventually implement this. Maybe I can add text here to clarify that, but removing it would leave the prototype incomplete. Interested in @ChumpChief's thoughts on whether we should remove this endpoint from the docs?

Comment (ChumpChief) [reply examples/hosts/app-integration/external-data/README.md:42]: Maybe drop the "echo" info (yes we'll use it to echo internally, but the endpoint doesn't need to reflect that externally) and make it something like

```suggestion
2. POST `/external-data-webhook`: Registered to be called by the External Data Service webhook when there's been a change to data upstream.
```

And then the "echo webhook" section of the documentation can cover the internal behavior of echoing the webhook.

Comment (Josmithr) [reply examples/hosts/app-integration/external-data/README.md]: I guess I'm fine with leaving docs about it if we want to describe the full flow, but we should note that this is a route established by the customer service to be used exclusively as a subscription endpoint for the external data service's webhook, and should be considered a private implementation detail to the service.

**Review by Josmithr — APPROVED** [review]
🚀
